src/common-utils/main.sh

```diff
@@ -0,0 +1,492 @@
+        if [ ! -e "${user_rc_path}/${rc_file}" ] || [ -s "${user_rc_path}/${rc_file}" ]; then
+            cp "/etc/skel/${rc_file}" "${user_rc_path}/${rc_file}"
+            chown ${USERNAME}:${group_name} "${user_rc_path}/${rc_file}"
+        fi
+    fi
+done
+
+# Add RC snippet and custom bash prompt
+if [ "${RC_SNIPPET_ALREADY_ADDED}" != "true" ]; then
+    case "${ADJUSTED_ID}" in
+        "debian")
+            global_rc_path="/etc/bash.bashrc"
+            ;;
+        "rhel")
+            global_rc_path="/etc/bashrc"
+            ;;
+        "alpine")
+            global_rc_path="/etc/bash/bashrc"
+            ;;
+    esac
+    cat "${FEATURE_DIR}/scripts/rc_snippet.sh" >> ${global_rc_path}
+    cat "${FEATURE_DIR}/scripts/bash_theme_snippet.sh" >> "${user_rc_path}/.bashrc"
+    if [ "${USERNAME}" != "root" ]; then
+        cat "${FEATURE_DIR}/scripts/bash_theme_snippet.sh" >> "/root/.bashrc"
+        chown ${USERNAME}:${group_name} "${user_rc_path}/.bashrc"
+    fi
+    RC_SNIPPET_ALREADY_ADDED="true"
+fi
+
+# Optionally configure zsh and Oh My Zsh!
+if [ "${INSTALL_ZSH}" = "true" ]; then
+    if [ "${ZSH_ALREADY_INSTALLED}" != "true" ]; then
+        if [ "${ADJUSTED_ID}" = "rhel" ]; then
+             global_rc_path="/etc/zshrc"
+        else
+            global_rc_path="/etc/zsh/zshrc"
+        fi
+        cat "${FEATURE_DIR}/scripts/rc_snippet.sh" >> /etc/zshrc
+        ZSH_ALREADY_INSTALLED="true"
+    fi
+
+    if [ "${CONFIGURE_ZSH_AS_DEFAULT_SHELL}" == "true" ]; then
+        chsh --shell /bin/zsh ${USERNAME}
+    fi
+
+    # Adapted, simplified inline Oh My Zsh! install steps that adds, defaults to a codespaces theme.
+    # See https://github.com/ohmyzsh/ohmyzsh/blob/master/tools/install.sh for official script.
+    oh_my_install_dir="${user_rc_path}/.oh-my-zsh"
+    if [ ! -d "${oh_my_install_dir}" ] && [ "${INSTALL_OH_MY_ZSH}" = "true" ]; then
+        template_path="${oh_my_install_dir}/templates/zshrc.zsh-template"
+        user_rc_file="${user_rc_path}/.zshrc"
+        umask g-w,o-w
+        mkdir -p ${oh_my_install_dir}
+        git clone --depth=1 \
+            -c core.eol=lf \
+            -c core.autocrlf=false \
+            -c fsck.zeroPaddedFilemode=ignore \
+            -c fetch.fsck.zeroPaddedFilemode=ignore \
+            -c receive.fsck.zeroPaddedFilemode=ignore \
+            "https://github.com/ohmyzsh/ohmyzsh" "${oh_my_install_dir}" 2>&1
+        echo -e "$(cat "${template_path}")\nDISABLE_AUTO_UPDATE=true\nDISABLE_UPDATE_PROMPT=true" > ${user_rc_file}
+        sed -i -e 's/ZSH_THEME=.*/ZSH_THEME="devcontainers"/g' ${user_rc_file}
+
+        # Add Dev Containers theme
+        mkdir -p ${oh_my_install_dir}/custom/themes
+        cp -f "${FEATURE_DIR}/scripts/devcontainers.zsh-theme" "${oh_my_install_dir}/custom/themes/devcontainers.zsh-theme"
+        ln -s "${oh_my_install_dir}/custom/themes/devcontainers.zsh-theme" "${oh_my_install_dir}/custom/themes/codespaces.zsh-theme"
```

Comment:
> Finally, the theme was renamed "devcontainers" rather than codespaces, though a symlink is in place for back compat.

I understand that the renamed theme has a back compat for codespaces. I don't believe this would impact codepsaces. However, would just like to double check if Codespaces might be impacted in any way due to this?

Reply:
Nope.  Both the name "devcontainer" and "codespace" for the theme will work. So existing dotfiles wouldn't be affected either if they referene the existing id.

Reply:
Sounds good 👍